src/mesh/interpolation.cxx

```diff
@@ -222,8 +228,134 @@ const Field3D interp_to(const Field3D &var, CELL_LOC loc)
+
+    Field2D result(var.getMesh());
+
+    if (region != RGN_NOBNDRY) {
+      // result is requested in some boundary region(s)
+      result = var; // NOTE: This is just for boundaries. FIX!
+    }
+    result.allocate();
+
+    // Cell location of the input field
+    CELL_LOC location = var.getLocation();
+    
+    if((location == CELL_CENTRE) || (loc == CELL_CENTRE)) {
+      // Going between centred and shifted
+      
+      stencil s;
+      CELL_LOC dir; 
+      
+      // Get the non-centre location for interpolation direction
+      dir = (loc == CELL_CENTRE) ? location : loc;
+
+      switch(dir) {
+      case CELL_XLOW: {
+        for(const auto &i : result.region(RGN_NOBNDRY)) {
+
+	  // Set stencils
+	  s.c = var[i];
+	  s.p = var[i.xp()];
+	  s.m = var[i.xm()];
+	  s.pp = var[i.offset(2,0,0)];
+	  s.mm = var[i.offset(-2,0,0)];
+	  
+	  if ((location == CELL_CENTRE) && (loc == CELL_XLOW)) {
+	    // Producing a stencil centred around a lower X value
+	    s.pp = s.p;
+	    s.p  = s.c;
+	  } else if (location == CELL_XLOW) {
+	    // Stencil centred around a cell centre
+	    s.mm = s.m;
+	    s.m  = s.c;
+	  }
+
+	  result[i] = interp(s);
+	}
+        break;
+      }
+      case CELL_YLOW: {
+        
+        if (mesh->ystart > 1) {
+          // More than one guard cell, so set pp and mm values
+          // This allows higher-order methods to be used
+          for(const auto &i : result.region(RGN_NOBNDRY)) {
+            // Set stencils
+            s.c = var[i];
+            s.p = var[i.yp()];
+            s.m = var[i.ym()];
+            s.pp = var[i.offset(0,2,0)];
+            s.mm = var[i.offset(0,-2,0)];
+            
+            if ((location == CELL_CENTRE) && (loc == CELL_YLOW)) {
+              // Producing a stencil centred around a lower Y value
+              s.pp = s.p;
+              s.p  = s.c;
+            } else if(location == CELL_YLOW) {
+              // Stencil centred around a cell centre
+              s.mm = s.m;
+              s.m  = s.c;
+            }
+            
+            result[i] = interp(s);
+          }
+        } else {
+          // Only one guard cell, so no pp or mm values
+          s.pp = nan("");
+          s.mm = nan("");
+          for(const auto &i : result.region(RGN_NOBNDRY)) {
+            // Set stencils
+            s.c = var[i];
+            s.p = var[i.yp()];
+            s.m = var[i.ym()];
+            
+            if ((location == CELL_CENTRE) && (loc == CELL_YLOW)) {
+              // Producing a stencil centred around a lower Y value
+              s.pp = s.p;
+              s.p  = s.c;
+            } else if(location == CELL_YLOW) {
+              // Stencil centred around a cell centre
+              s.mm = s.m;
+              s.m  = s.c;
+            }
+            
+            result[i] = interp(s);
+          }
+        }
+	break;
+      }
+      default: {
+	// This should never happen
+	throw BoutException("Don't know what to do");
```

Comment:
Could this message be more explicit, e.g. unknown stagger direction in interp_to or similar?